src/GoogleStorageAdapter.php

```diff
@@ -354,4 +354,35 @@ protected function getObject($path)
     {
         return $this->service->objects->get($this->bucket, $path);
     }
+
+    /**
+     * Adds an ACL entry that makes the object world-readable
+     *
+     * @param string $path Object path in the current bucket
+     */
+    protected function publishObject($path) {
+        if ($this->getRawVisibility($path) === AdapterInterface::VISIBILITY_PUBLIC) {
+            return;
+        }
+
+        $publicAcl = new \Google_Service_Storage_ObjectAccessControl();
+        $publicAcl->setEntity('allUsers');
+        $publicAcl->setRole('READER');
+        $this->service->objectAccessControls->insert($this->bucket, $path, $publicAcl);
+    }
+
+    /**
+     * Removes a `READER` role for entity `allUsers` from the object if present.
+     *
+     * @param string $path
+     */
+    protected function unPublishObject($path) {
+        $controls = $this->service->objectAccessControls->listObjectAccessControls($this->bucket, $path);
+        // Cycle through existent entries in the ACL and only delete the `allUsers` entry if it is set to `allUsers`
+        foreach ($controls->getItems() as $control) {
+            if ($control['role'] == 'READER' && $control['entity'] == 'allUsers') {
+                $this->service->objectAccessControls->delete($this->bucket, $path, 'allUsers');
```

Comment:
Since we aren't using anything from the item that matches these criteria, can we break after the first match?